ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```diff
@@ -1,25 +1,67 @@
 ---
+# During HA control plane upgrade server address in kubeconfig is switched to local for
+#   * compatibility between client and server versions
+#   * identifying correct server version
+
+- name: k8s/master | Switch apiserver address to local
+  include_tasks: kubernetes/utils/set-local-apiserver.yml # sets kubectl_context_cluster
+  when:
+    - groups.kubernetes_master | length > 1
+    - inventory_hostname in groups.kubernetes_master
+
 - name: Wait for kube-apiserver and get cluster version
-  delegate_to: "{{ groups.kubernetes_master[0] }}"
+  delegate_to: >-
+    {{ inventory_hostname if inventory_hostname in groups.kubernetes_master else
+      groups.kubernetes_master[0] }}
   block:
     - name: k8s | Include wait-for-kube-apiserver.yml
       import_tasks: kubernetes/utils/wait-for-kube-apiserver.yml
 
     - name: k8s | Include get-cluster-version.yml
-      import_tasks: kubernetes/get-cluster-version.yml # sets cluster_version
+      import_tasks: kubernetes/get-cluster-version.yml
+
+    - name: Set initial_cluster_version fact # sets initial_cluster_version
+      set_fact:
+        initial_cluster_version: >-
+          {{ (kubectl_cluster_version.stdout | from_yaml).serverVersion.gitVersion }}
+
+    - name: k8s/kubelet-cm | Include set-cluster-version.yml
+      include_tasks: kubernetes/set-cluster-version.yml  # sets cluster_version
 
 - name: k8s | Include get-kubelet-version.yml
-  import_tasks: kubernetes/get-kubelet-version.yml # sets kubelet_version
+  import_tasks: kubernetes/get-kubelet-version.yml
   delegate_to: "{{ groups.kubernetes_master[0] }}"
 
+- name: Set kubelet version as fact
```

Comment:
As I see some tasks start with 'k8s |', and others aren't in the same place. Should be changed?

Reply:
Honestly I am not aware that's the convention here, need to clarify.

Reply:
I will change some of the following schema:
- if task includes or import another task for specific component I will add k8s
- if task is performing action on specific component/service I will add k8s
- if task is responsible for keeping the upgrade procedure ongoing, traces it and is a helper, I will leave it without k8s prefix
Do you agree with the proposed form?